### PR TITLE
username-and-password-6

### LINKED ISSUE #6: Add options for setting Fluidinfo username and password
Add options for setting Fluidinfo username and password. Code for this can be taken from the lastpage-chrome extension of @gridaphobe.

### PR DESCRIPTION
**Pull request by terrycojones**
gridaphobe: **approved**

Adds a way to enter your username and password, and a popup for tagging the page you're on (or using its URL as a tag value).

Fixes #6

### CODE REVIEW

File: popup.js [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+                status('Tag saved successfully.');
+            }
+            else {
+                ok = false;
+                status(response.message);
+            }
+            maybeRunCallback();
+        });
+    }
+    else if (tagName1.value){
```
Comment (gridaphobe) [inline popup.js:119]: The help text says the tag-value is optional, and yet you throw an error if it is left blank :-(

I would suggest removing the tagName1.value test and just let the `parseFluidinfoValue` code create a null tag.

File: popup.js [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+    }
+}
+
+function clearStatus(msg){
+    var s = document.getElementById('_fi_status');
+    s.innerText = '';
+    s.style.visibility = 'hidden';
+}
+
+function save(url, callback){
```
Comment (gridaphobe) [inline popup.js:73]: This function seems to work as advertised, but I don't see why it should have to (potentially) make so many api calls. Why not just construct a single payload to pass to a PUT /values? Then you don't have to worry about tracking the # of calls made :-)

File: background.js [hunk 3]
```diff
@@ -101,27 +71,68 @@ chrome.contextMenus.create({
...
+        'contexts' : ['link'],
+        'onclick' : function(info, tab){
+            openNewTab(currentLowercaseLinkText, info, tab);
+        }
+    });
+};
+
+// Listen for incoming messages with new link text, and update our currentLinkText
+// variable as well as the context menu title for the link text. Add a context menu
+// item for the lowercase string too, if it's not the same as the link text.
```
Comment (gridaphobe) [inline background.js:104]: I understand the reasoning for lowercasing the about value, but I think it looks bad to display both standard case and lowercased strings. I'd prefer just showing the lowercased version even though I don't particularly like lowercasing in general.

File: popup.js (hunk 1 above)
Comment (terrycojones) [inline popup.js:119]: Yes, you're right, it should be different. I'll fix it on this branch/ticket https://github.com/fluidinfo/obrowser-chrome-extension/issues/10

**Comment by gridaphobe** [comment]
Overall I think it looks good. It might be nice to have the read-later/like checkboxes pre-checked if the tags are already present, that would also allow users to remove the boolean tags from within the extension. Also, I'm not sure if/how you can do this, but clicking the icon while the popup is showing should close the popup.

File: popup.js (hunk 2 above)
Comment (terrycojones) [inline popup.js:73]: Yeah.... short answer is that it just happened that way :-)

Slightly better answer(?), if you're using the URL value as a tag value as well as tagging the URL itself, then you still have to make at least 2 calls.

Anyway, you're right that all the calls using the URL as an about value could be collapsed. In the future :-)

File: background.js (hunk 3 above)
Comment (terrycojones) [inline background.js:104]: Yeah too.  I think this should become an option...  also in the future :-)

**Comment by gridaphobe** [comment]
Ok then, approved with one last comment. The "advanced mode" box in the options page doesn't seem to do anything here. Is that correct? If so just remove it as part of this pull req.

**Comment by terrycojones** [comment]
Yes, I want to do that too, to populate the form - at least as much as I can. That's going to be slightly tricky since we can't display all possible tag value types. I think it's important, but I also don't want to recreate the entire object browser in the popup :-)  So we need to figure out a balance.

For me, clicking the icon (I mean the extension icon in the browser toolbar) when the popup is open does close it. I don't thing there's any other way (I mean under my control) to close it. I added a cancel button at some point, but it had no effect.

**Comment by terrycojones** [comment]
Agreed, advanced mode (at least in the UI) should vanish. Now!

**Pull request merged**